src/app/home/home.component.ts

```diff
@@ -18,12 +19,17 @@ export class HomeComponent implements OnInit {
   }
 
   editData(data:any,value:any){
-    
+    this.service.setIndex(value)
     this.service.setItem(this.userData[value])
     this.router.navigateByUrl('');
   }
+  
   deleteData(data:any,value:any){
+    this.userData.splice(value,1)
+    localStorage.setItem('user',JSON.stringify(this.userData))
+  }
 
-
+  back(){
+    this.router.navigateByUrl('')
```

Comment:
Why are we redirecting to blank url, u can use go back function to navigate back